woocommerce/woocommerce
# Moving currencyContext to currency package and updating references

### Changes proposed in this Pull Request:

This is a prerequisite to [this issue](https://github.com/woocommerce/woocommerce/issues/36395), which involves moving a number of the product fields into the `product-editor` package. One of the dependencies for these components is the `currencyContext`, which would suit the `@woocommerce/currency` package better anyways. 

Since doing this all within the same PR is touching a great many files, I'm separating this out to it's own PR. 

Closes part of https://github.com/woocommerce/woocommerce/issues/36395 .



### How to test the changes in this Pull Request:



**Note that this is an extremely small tweak, but it touches a lot of files, so a quick smoke test of several areas is likely needed**
1. Checkout branch and open JS console
2. Run through the onboarding wizard and ensure currencies display properly and there are no relevant errors thrown.
3. Use smooth generator to add several dummy products and orders.
4. Quick smoke test of homescreen and ensure currencies display properly and there are no relevant errors thrown.
5. Quick smoke test of Analytics reports and ensure currencies display properly and there are no relevant errors thrown.



### Other information:




### FOR PR REVIEWER ONLY:
